### PR TITLE
SDK Style Projects

### PR DESCRIPTION
**Pull request by TheBlubb14**
Converts all project to the new sdk style project types. This should simplify the actual step to .Net 6 and making a working CI.
Also we can now build the projects with `dotnet` 🥳 

The installer project was not converted as I do not have the proper tooling installed

### CODE REVIEW

**Comment by TheBlubb14** [comment]
I just took the assembly versions for each project from the matching `AssemblyInfo.cs`. So the projects have each a different version. Maybe they need to be sync in future

**Comment by tomwendel** [comment]
cooool 🥰

**Comment by tomwendel** [comment]
> I just took the assembly versions for each project from the matching `AssemblyInfo.cs`. So the projects have each a different version. Maybe they need to be sync in future

Yes. Thats what I wanted to get automated by git action. I always set version manually in the past 😔

**Comment by TheBlubb14** [comment]
> > I just took the assembly versions for each project from the matching `AssemblyInfo.cs`. So the projects have each a different version. Maybe they need to be sync in future
> 
> Yes. Thats what I wanted to get automated by git action. I always set version manually in the past 😔

For the CI you can easily pass the version like this `dotnet build -p:Version=99.0.1`

**Comment by tomwendel** [comment]
> > > I just took the assembly versions for each project from the matching `AssemblyInfo.cs`. So the projects have each a different version. Maybe they need to be sync in future
> > 
> > Yes. Thats what I wanted to get automated by git action. I always set version manually in the past 😔
> 
> For the CI you can easily pass the version like this `dotnet build -p:Version=99.0.1`

good to know! now we need a proper way to calculate version numbers. is there any kind of convention?

**Comment by TheBlubb14** [comment]
> > > > I just took the assembly versions for each project from the matching `AssemblyInfo.cs`. So the projects have each a different version. Maybe they need to be sync in future
> > > 
> > > 
> > > Yes. Thats what I wanted to get automated by git action. I always set version manually in the past 😔
> > 
> > 
> > For the CI you can easily pass the version like this `dotnet build -p:Version=99.0.1`
> 
> good to know! now we need a proper way to calculate version numbers. is there any kind of convention?

I have used this in an earlier project https://github.com/anothrNick/github-tag-action
Example: https://github.com/TheBlubb14/SpotifyNet/blob/master/.github/workflows/main.yml

File: SharedComponents/AntVideo/Block/FolkLost.cs [hunk 1]
```diff
@@ -1,32 +0,0 @@
...
-
-        public override void Serialize(Stream outputstream) {
-            base.Serialize(outputstream);
-            Serializer.SendUshort(outputstream, folkId);
-        }
-
-        public new void Deserialize(Stream stream) {
-            folkId = Serializer.ReadUShort(stream);
-        }
-
```
Comment (tomwendel) [inline SharedComponents/AntVideo/Block/FolkLost.cs:29]: das ist ne schlechte Idee. Das hier ist teil des AntVideo Protokolls. Mag sein, dass es nicht verwendet wird aktuell, aber das Protokoll erlaubt ein verschwindendes Volk mitten in der Simulation.

Comment (tomwendel) [reply SharedComponents/AntVideo/Block/FolkLost.cs:29]: Kommando zurück. Schau mir gerade das Protokoll an und sehe kein "FolkLost". Entweder es gibt einen Namensmissmatch zwischen Enum und Klassen (Folk vs. Colony vs. Team) oder das Ding war wirklich eine tote Altlast...?

**Review by tomwendel — APPROVED** [review]
Puh. Da sind ja noch jede Menge Altlasten am Start. Saucool.

**Pull request merged**